Added Automatic Installation of Infernal Resources

Dariensg opened the pull request:
-Added to ClientSetup to copy Infernal Resources to MC Resource Pack folder
-Fixed the directory of the textures in the resource pack zip

Dariensg committed:
Added Automatic Installation of Infernal Resources

-Added to ClientSetup to copy Infernal Resources to MC Resource Pack folder
-Fixed the directory of the textures in the resource pack zip

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/InfernalExpansion.java:
```diff
@@ -208,6 +212,32 @@ protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
+                InputStream in = InfernalExpansion.class.getResourceAsStream("/assets/infernalexp/infernal_resources.zip");
+                FileOutputStream out = new FileOutputStream(target);
+
+                byte[] buf = new byte[16384];
```
What is this magic number `16384`? Please document it with comments at least.

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/InfernalExpansion.java:
```diff
@@ -208,6 +212,32 @@ protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
+
+    private static void loadInfernalResources() {
+        // Creates file location for resource pack
+        File dir = new File(".", "resourcepacks");
```
I believe there is a method for retrieving the working directory for Minecraft, it should be used instead of the `.` directory.

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/InfernalExpansion.java:
```diff
@@ -208,6 +212,32 @@ protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
     private void clientSetup(final FMLClientSetupEvent event) {
         DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> InfernalExpansionClient::init);
+
+        loadInfernalResources();
```
The function call should be in `InfernalExpansionClient::init`, not here.

Dariensg committed:
Moved loadInfernalResources

-Moved loadInfernalResources to InfernalExpansionClient
-Added more clear comments about the buffer for reading the InputStream

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/client/InfernalExpansionClient.java:
```diff
@@ -35,5 +41,34 @@ public static void init() {
 		ItemModelsProperties.registerProperty(IEItems.BLINDSIGHT_TONGUE_WHIP.get(), new ResourceLocation("attacking"), (itemStack, clientWorld, livingEntity) -> livingEntity != null && (((IWhipItem) itemStack.getItem()).getAttacking() || ((IWhipItem) itemStack.getItem()).getCharging()) && livingEntity.getHeldItemMainhand() == itemStack ? 1.0F : 0.0F);
-	}
+
+        loadInfernalResources();
```
Would've preferred `InfernalExpansionClient.loadInfernalResources()`, just so you know for future reference.

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/client/InfernalExpansionClient.java:
```diff
@@ -35,5 +41,34 @@ public static void init() {
+                out.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
```
```suggestion
        if (!target.exists()) {
            try {
                dir.mkdirs();
                InputStream in = InfernalExpansion.class.getResourceAsStream("/assets/infernalexp/infernal_resources.zip");
                FileOutputStream out = new FileOutputStream(target);

                // The number of bytes here is how many can be read from the resource pack at one time
                // 16K is standard for reading from disk and should not be tampered with
                byte[] buf = new byte[16384];
                int len = 0;
                while((len = in.read(buf)) > 0)
                    out.write(buf, 0, len);

                in.close();
                out.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
```

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/client/InfernalExpansionClient.java:
```diff
@@ -35,5 +41,34 @@ public static void init() {
+                FileOutputStream out = new FileOutputStream(target);
+
+                // The number of bytes here is how many can be read from the resource pack at one time
+                // 16K is standard for reading from disk and should not be tampered with
```
```suggestion
                // 16kB is the most common disk chunk size, and using this array size
                // reduces latency between reading and actually processing the data.
                // The performance difference is not significant, but it's improved by using a 16kB array.
```

SwanX1 reviewed (changes requested):
Minor formatting issues.

Dariensg committed:
Apply suggestions from code review

Co-authored-by: Kārlis Čerņavskis <karliscern@gmail.com>

Dariensg committed:
Update src/main/java/com/nekomaster1000/infernalexp/client/InfernalExpansionClient.java

Co-authored-by: Kārlis Čerņavskis <karliscern@gmail.com>

Dariensg committed:
Formatting Fix

-Fixed call to static method through class

SwanX1 reviewed (approved):
LGTM, merge at your discretion, if you have nothing else to change. If you do, request a new review from me.

SwanX1 commented:
Actually, message me on discord if you're okay with this being merged, it has a conflict that must be resolved, and I'll bump the version right after.
